src/shared/awsClientBuilder.ts

```diff
@@ -3,11 +3,21 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { Request, AWSError } from 'aws-sdk'
 import { ServiceConfigurationOptions } from 'aws-sdk/lib/service'
 import { env, version } from 'vscode'
 import { AwsContext } from './awsContext'
 import { extensionVersion } from './vscode/env'
 
+interface RequestExtras {
+    readonly service: AWS.Service
+    readonly operation: string
+    readonly params?: any
+}
+
+type RequestListener = (request: AWS.Request<any, AWSError> & RequestExtras) => void
```

Comment:
this type could be inlined. Naming things uses up the conceptual space and implies to human readers that the named thing is a recurring concept. Then we end up with many similar-named things that are different structurally or even semantically, which to me suggests that they are anonymous.

Reply:
It'd be nice if type alias names weren't exposed outside the scope of the file unless they're exported. But right now it does seem like a new "thing" is being defined when in reality it's just an alias to stop code (type?) duplication.